pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -102,13 +102,14 @@ def log_metrics(self, metrics: _OUT_DICT, step: Optional[int] = None) -> None:
         if self.trainer.logger is None or not metrics:
             return
 
-        # turn all tensors to scalars
-        scalar_metrics = metrics_to_scalars(metrics)
+        if step is None and "step" in metrics:
```

Comment:
I don't get why the step changes. The step is supposed to be an int in all cases.

This should only change what we update with:

```diff
        # turn all tensors to scalars
        scalar_metrics = metrics_to_scalars(metrics)

        if step is None:
            step = scalar_metrics.pop("step", None)

-        self._logged_metrics.update(scalar_metrics)
+        self._logged_metrics.update(metrics)
```

Reply:
it's typed with optional[int], and we don't keep step inside logged_metrics so we need to extract it from metrics itself before updating the logged_metrics.
was getting the mypy error: https://github.com/PyTorchLightning/pytorch-lightning/runs/4645001501?check_suite_focus=true

Reply:
My suggested change does not output any mypy errors locally.

And we don't need to remove `step` from `metrics` as it would be there only if added by the user.

Reply:
okay.. will update. it will a minor change in the behavior then.